src/Services/InventoryUpdateService.php

```diff
@@ -322,34 +412,112 @@ public function getFilters(bool $fullInventory): array
   /**
    * Get the stock buffer value, normalized to 0
    *
-   * @param AbstractConfigHelper $configHelper
-   * @param LoggerContract $loggerContract
    * @return int
    */
-  private static function getNormalizedStockBuffer($configHelper, $loggerContract = null)
+  private function getNormalizedStockBuffer()
   {
     $stockBuffer = null;
-    if (isset($configHelper)) {
-      $stockBuffer = $configHelper->getStockBufferValue();
-    }
+    $stockBuffer = $this->configHelper->getStockBufferValue();
+
 
     if (isset($stockBuffer)) {
       if ($stockBuffer >= 0) {
         return $stockBuffer;
       }
 
       // invalid value for buffer
-      if (isset($loggerContract)) {
-        $loggerContract->warning(
-          TranslationHelper::getLoggerKey(self::LOG_KEY_INVALID_STOCK_BUFFER),
-          [
-            'additionalInfo' => ['stockBuffer' => $stockBuffer],
-            'method' => __METHOD__
-          ]
-        );
-      }
+      $this->logger->warning(
+        TranslationHelper::getLoggerKey(self::LOG_KEY_INVALID_STOCK_BUFFER),
+        [
+          'additionalInfo' => [
+            'stockBuffer' => json_encode($stockBuffer)
+          ],
+          'method' => __METHOD__
+        ]
+      );
     }
 
     return 0;
   }
+
+  /**
+   * Check if the state of the Inventory Service has been "running" for more than the maximum allotted time
+   * This functionality was extracted from the old UpdateFullInventoryStatusCron
+   * @return bool
+   */
+  private function serviceHasBeenRunningTooLong(): bool
+  {
+    if (!$this->statusService->isInventoryRunning()) {
+      return false;
+    }
+
+    $currentSessionStartedAt = $this->statusService->getStartOfMostRecentAttempt();
+    if (isset($currentSessionStartedAt) && !empty($currentSessionStartedAt) && (time() - strtotime($currentSessionStartedAt)) > self::MAX_INVENTORY_TIME) {
+
+      $this->logger->warning(TranslationHelper::getLoggerKey(self::LOG_KEY_LONG_RUN), [
```

Comment:
Can we send this to external log as well?

Reply:
This might cause too much log noise.

Reply:
as long as we log it properly and kibana can search/filter, too much log should not be a problem.
this log may be helpful for us to keep an eye on the sync performance since we change the way to sync inventory.
If you think that is too much log noise, I am fine too.